test/functional/test_framework/wallet.py

```diff
@@ -72,8 +72,9 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
         tx_hex = tx.serialize().hex()
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
-        self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
+        utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
         from_node.sendrawtransaction(tx_hex)
         assert_equal(tx_info['vsize'], vsize)
         assert_equal(tx_info['fees']['base'], fee)
+        self._utxos = utxos  # update our internal utxo set after we are sure send_self_transfer will succeed
```

Comment:
nit: Aren't we assigning utxos when `sendrawtransaction` succeeds? So, shouldn't it be `#update... after we are sure that sendrawtransaction will succeed` because in the case if we add sign_dont_send options, the `send_self_transfer` succeeds but dont spend tx?

Reply:
Good point - I improved the comment. Pushed and rebased